led_test.go

```diff
@@ -0,0 +1,395 @@
+	}
+	return len(b), nil
+}
+
+// Size returns the length of the backing data and a nil error.
+func (l *ledBrightness) Size() (int64, error) {
+	return size(l.brightness), nil
+}
+
+// ledDelayOn is the delay_on attribute.
+type ledDelayOn led
+
+// ReadAt satisfies the io.ReaderAt interface.
+func (l *ledDelayOn) ReadAt(b []byte, offset int64) (int, error) {
+	return readAt(b, offset, l.delayOn)
+}
+
+// Truncate is a no-op.
+func (l *ledDelayOn) Truncate(_ int64) error { return nil }
+
+// WriteAt satisfies the io.WriterAt interface.
+func (l *ledDelayOn) WriteAt(b []byte, off int64) (int, error) {
+	i, err := strconv.Atoi(string(chomp(b)))
+	if err != nil {
+		l.t.Errorf("unexpected error: %v", err)
+		return len(b), syscall.EINVAL
+	}
+	l.delayOn = i
+	return len(b), nil
+}
+
+// Size returns the length of the backing data and a nil error.
+func (l *ledDelayOn) Size() (int64, error) {
+	return size(l.delayOn), nil
+}
+
+// ledDelayOff is the delay_off attribute.
+type ledDelayOff led
+
+// ReadAt satisfies the io.ReaderAt interface.
+func (l *ledDelayOff) ReadAt(b []byte, offset int64) (int, error) {
+	return readAt(b, offset, l.delayOff)
+}
+
+// Truncate is a no-op.
+func (l *ledDelayOff) Truncate(_ int64) error { return nil }
+
+// WriteAt satisfies the io.WriterAt interface.
+func (l *ledDelayOff) WriteAt(b []byte, off int64) (int, error) {
+	i, err := strconv.Atoi(string(chomp(b)))
+	if err != nil {
+		l.t.Errorf("unexpected error: %v", err)
+		return len(b), syscall.EINVAL
+	}
+	l.delayOff = i
+	return len(b), nil
+}
+
+// Size returns the length of the backing data and a nil error.
+func (l *ledDelayOff) Size() (int64, error) {
+	return size(l.delayOff), nil
+}
+
+// ledTrigger is the trigger attribute.
+type ledTrigger led
+
+// ReadAt satisfies the io.ReaderAt interface.
+func (l *ledTrigger) ReadAt(b []byte, offset int64) (int, error) {
+	return readAt(b, offset, l)
+}
+
+// Truncate is a no-op.
+func (l *ledTrigger) Truncate(_ int64) error { return nil }
+
+// WriteAt satisfies the io.WriterAt interface.
+func (l *ledTrigger) WriteAt(b []byte, off int64) (int, error) {
+	set := string(chomp(b))
+	for k := range l.trigger {
+		l.trigger[k] = false
+	}
+	l.trigger[set] = true
+	return len(b), nil
+}
+
+// Size returns the length of the backing data and a nil error.
+func (l *ledTrigger) Size() (int64, error) {
+	return size(l), nil
+}
+
+func (l *ledTrigger) String() string {
+	s := make([]string, 0, len(l.trigger))
+	for k, set := range l.trigger {
+		if set {
+			s = append(s, fmt.Sprintf("[%s]", k))
+			continue
+		}
+		s = append(s, k)
+	}
+	sort.Strings(s)
+	return strings.Join(s, " ")
+}
+
+// ledUevent is the uevent attribute.
+type ledUevent led
+
+// ReadAt satisfies the io.ReaderAt interface.
+func (l *ledUevent) ReadAt(b []byte, offset int64) (int, error) {
+	return readAt(b, offset, l)
+}
+
+// Size returns the length of the backing data and a nil error.
+func (l *ledUevent) Size() (int64, error) {
+	return size(l), nil
```

Comment:
When the receiver is a fmt.Stringer it is passed, otherwise the field is passed. Whether it is a fmt.Stringer depends on whether the field would render correctly when printed via a `%v` fmt verb; numbers and strings are fine like this, but things that depend on multiple fields or have complex formatting (e.g. trigger) aren't.